src/analysis/processing/qgsalgorithmsumlinelength.cpp

```diff
@@ -131,6 +125,7 @@ bool QgsSumLineLengthAlgorithm::prepareAlgorithm( const QVariantMap &parameters,
   {
     mDa.setEllipsoid( context.project()->ellipsoid() );
   }
+  mCrs = mLinesSource->sourceCrs();
```

Comment:
This isn't correct -- we need to use the polygon's source crs here, which may differ from the line's source crs. 

I think the solution will be to defer creation of mDa until the first call to processFeature, at which stage we can be certain that mCrs has correctly been set to match the input polygon's crs.

Reply:
Oops! That's right, it should be polygons' Crs. 

Well, a couple of possible solutions here:

1. Inside `processFeature()`, check if `mDa` is set already, if not, set it. (Is this what you suggested in your comment?)
2. Inside `prepareAlgorithm()`, get the polygon (`INPUT`) parameter and use its Crs to set mDa's source Crs. I think it's more natural to have everything set in this method.